sendgrid.go

```diff
@@ -21,6 +23,10 @@ func GetRequest(key, endpoint, host string) rest.Request {
 // GetRequestSubuser like GetRequest but with On-Behalf of Subuser
 // @return [Request] a default request object
 func GetRequestSubuser(key, endpoint, host, subuser string) rest.Request {
+	if strings.Contains(endpoint, "v3/mail/send") {
```

Comment:
Does using the `on-behalf-of` with this endpoint throw any error at all? I'm not too sure I like hardcoding this into the logic here.

Reply:
It does not throw an error, it just silently "fails" by ignoring the header and putting the email on the subuser who owns the API key.

Reply:
(My primary motivation for this change was so that there would be something to test to guard against someone adding this functionality back in)

Reply:
I am ok with getting rid of `NewSendClientSubuser` and updating the documentation since that's not really supported but I don't like adding this hardcoded panic.

Reply:
@shwetha-manvinkurke Done